express-notes-api/index.js

```diff
@@ -0,0 +1,96 @@
+var express = require('express');
+const dataJson = require('./data.json');
+const fs = require('fs');
+var app = express();
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.use(express.json());
+
+// Client can GET list of notes
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const key in dataJson.notes) {
+    notesArray.push(dataJson.notes[key]);
+  }
+  res.json(notesArray);
+});
+
+// Client can GET notes by specific id
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!Number.isInteger(id) || id < 0) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (dataJson.notes[id]) {
+    res.status(200).json(dataJson.notes[id]);
+  } else {
+    res.status(404).json({ error: `cannot find note with id ${id}` });
+  }
+});
+
+// Client can POST new note
+app.post('/api/notes', (req, res) => {
+  const newEntry = req.body;
+  if (!('content' in newEntry)) {
+    res.status(404).json({ error: 'content is a required field' });
+  } else if ('content' in newEntry) {
+    newEntry.id = dataJson.nextId;
+    dataJson.notes[dataJson.nextId] = newEntry;
+    dataJson.nextId++;
+    const data = JSON.stringify(dataJson, null, 2);
+    fs.writeFile('data.json', data, err => {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ error: 'An unexpected error occurred.' });
+      }
+      res.status(201).json(newEntry);
```

Comment:
Here we'll want to include an `else` statement to ensure that the server can't accidentally try to respond twice to the same request.